src/wire/chunked_report.rs

```diff
@@ -0,0 +1,469 @@
+        if len < (field::PAYLOAD.start + n_chunk_payload_bytes) {
+            Err(ChunkedReportWireError::IncompletePayload)
+        } else if n_chunk_payload_bytes > Self::MAX_PAYLOAD_BYTES_PER_CHUNK {
+            Err(ChunkedReportWireError::TooManyPayloadBytes)
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Consumes the chunked report, returning the underlying buffer
+    pub fn into_inner(self) -> T {
+        self.buffer
+    }
+
+    /// Return the length of a chunked report header
+    pub fn header_len() -> usize {
+        field::PAYLOAD.start
+    }
+
+    /// Return the length of a buffer required to hold a chunked report
+    /// with a payload length of `n_chunk_payload_bytes`
+    pub fn buffer_len(n_chunk_payload_bytes: usize) -> usize {
+        field::PAYLOAD.start + n_chunk_payload_bytes
+    }
+
+    /// Return the length of the chunked report payload
+    pub fn payload_len(&self) -> usize {
+        self.n_chunk_payload_bytes() as usize
+    }
+
+    /// Return the `fingerprint` field
+    #[inline]
+    pub fn fingerprint(&self) -> u32 {
+        let data = self.buffer.as_ref();
+        LittleEndian::read_u32(&data[field::FINGERPRINT])
+    }
+
+    /// Return the `probe_id` field
+    #[inline]
+    pub fn probe_id(&self) -> u32 {
+        let data = self.buffer.as_ref();
+        LittleEndian::read_u32(&data[field::PROBE_ID])
+    }
+
+    /// Return the `chunk_group_id` field
+    #[inline]
+    pub fn chunk_group_id(&self) -> u16 {
+        let data = self.buffer.as_ref();
+        LittleEndian::read_u16(&data[field::CHUNK_GROUP_ID])
+    }
+
+    /// Return the `chunk_index` field
+    #[inline]
+    pub fn chunk_index(&self) -> u16 {
+        let data = self.buffer.as_ref();
+        LittleEndian::read_u16(&data[field::CHUNK_INDEX])
+    }
+
+    /// Return the `payload_data_type` field
+    #[inline]
+    pub fn payload_data_type(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::PAYLOAD_DATA_TYPE]
+    }
+
+    /// Return the `is_last_chunk` field
+    #[inline]
+    pub fn is_last_chunk(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::IS_LAST_CHUNK]
+    }
+
+    /// Return the `reserved` field
+    #[inline]
+    pub fn reserved(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::RESERVED]
+    }
+
+    /// Return the `n_chunk_payload_bytes` field
+    #[inline]
+    pub fn n_chunk_payload_bytes(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::N_CHUNK_PAYLOAD_BYTES]
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> ChunkedReport<&'a T> {
+    /// Return a pointer to the payload
+    #[inline]
+    pub fn payload(&self) -> &'a [u8] {
+        let data = self.buffer.as_ref();
+        &data[field::PAYLOAD]
+    }
+}
+
+impl<T: AsRef<[u8]> + AsMut<[u8]>> ChunkedReport<T> {
+    /// Set the `fingerprint` field
+    #[inline]
+    pub fn set_fingerprint(&mut self, value: u32) {
```

Comment:
Similar recommendation re: not permitting possible-invalid fingerprints to be set.